client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx

```diff
@@ -207,27 +210,23 @@ const InvestigationTable: React.FC = (): JSX.Element => {
                     />
                 );
             case TableHeadersNames.investigatorName:
-                if (selectedRow === indexedRow.epidemiologyNumber && shouldOpenInvestigatorAllocation) {
-                    return (
-                        <InvestigatorAllocationDialog
-                            isOpen={shouldOpenInvestigatorAllocation}
-                            setIsOpen={setShouldOpenInvestigatorsAllocation}
-                            investigators={getFilteredUsersOfCurrentCounty()}
-                            allocateInvestigationToInvestigator={allocateInvestigationToInvestigator}
-                            groupId={indexedRow.groupId}
-                            epidemiologyNumber={indexedRow.epidemiologyNumber}
-                        />
-                    )
-                } else {
                     return (
-                        <InvestigatorAllocationCell
-                            investigatorName={indexedRow[cellName as keyof typeof TableHeadersNames]}
-                            epidemiologyNumber={indexedRow.epidemiologyNumber}
-                            epiNumOnInvestigatorNameHover={epiNumOnInvestigatorNameHover}
-                            setEpiNumOnInvestigatorNameHover={setEpiNumOnInvestigatorNameHover}
-                        />
+                        <>
+                            { selectedRow === indexedRow.epidemiologyNumber && isInvestigatorAllocationDialogOpen && 
```

Comment:
I meant that you can put the InvestigatorAllocationDialog outside of this function in the InvestigationTable render area. 
Right now you have a condition for rendering the dialog but you also have an `isOpen` property which hides it. 
If you remove the condition for rendering the dialog you have a dialog in the dom for each cell in the table. 
If there are 100 investigations there will be 100 dialogs hidden in the dom.

Reply:
We should have one of this dialog for all the investigations in the table